File: Leetcode/Deepest Leaves Sum.java
// https://leetcode.com/problems/deepest-leaves-sum/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        if(root == null){
            return 0;
        }
        int h = height(root);
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int i = 1;
        int sum = 0;
        while(true){
            int size = q.size();
            if(size == 0){
                break;
            }
            while(size > 0){
                TreeNode temp = q.remove();
                if(i == h){
                    sum += temp.val;
                }
                if(temp.left != null){
                    q.add(temp.left);
                }
                if(temp.right != null){
                    q.add(temp.right);
                }
                size--;
            }
            i++;
        }
        return sum;
    }
    
    public int height(TreeNode root){
        int lst = 0, rst = 0;
        if(root == null){
            return 0;
        }
        lst = height(root.left);
        rst = height(root.right);
        return Math.max(lst, rst) + 1;
    }
}

File: Leetcode/Deepest Leaves Sum.py
import sys


def intInput(): return int(input())


def sepInput(): return map(int, input().split())


def listInput(): return list(map(int, input().split()))


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def __init__(self):

        self.su = 0

    def height(self, node):

        if node is None:

            return 0

        else:

            lheight = self.height(node.left)

            rheight = self.height(node.right)

            if lheight > rheight:

                return lheight + 1

            else:

                return rheight + 1

    def sumOfDeepestLeaves(self, root, h):

        if root is None:

            return

        elif h == 1:

            self.su += root.val

        else:

            self.sumOfDeepestLeaves(root.left, h - 1)

            self.sumOfDeepestLeaves(root.right, h - 1)

    def deepestLeavesSum(self, root) -> int:

        h = self.height(root)

        self.sumOfDeepestLeaves(root, h)

        return self.su
